extensions/api/data-management/asset/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/asset/service/AssetServiceImpl.java

```diff
@@ -54,6 +58,11 @@ public Asset findById(String assetId) {
 
     @Override
     public Collection<Asset> query(QuerySpec query) {
+        var result = queryValidator.validate(query);
+
+        if (result.failed()) {
+            throw new EdcQueryException(format("Error validating schema: %s", String.join(", ", result.getFailureMessages())));
```

Comment:
Now there's the `getFailureDetail` method on the `AbstractResult` class that already to the failure messages join

Reply:
ah, didn't know that.